tests/rptest/tests/compatibility/sarama_test.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2020 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import os
+import subprocess
+import time
+import requests
+from ducktape.mark.resource import cluster
+
+from rptest.services.compatibility.compat_producer import CompatProducer
+from rptest.services.compatibility.compat_example import CompatExample
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.clients.types import TopicSpec
+
+#Compatibility testing for Sarama
+class SaramaTest(RedpandaTest):
+    
+    TESTS_DIR = os.path.join("/root/go/src/github.com/Shopify", "sarama")
+    topics = (TopicSpec(), )
+
+    def __init__(self, test_context):
+        super(SaramaTest, self).__init__(test_context=test_context)
+        self._producer = CompatProducer(test_context, self.redpanda, self.topic)
+        self._example = CompatExample(test_context)
+
+    # runs sarama unit tests
+    def test_sarama(self):
+        #Form the command
+        cmd = f"cd {SaramaTest.TESTS_DIR} && go test -count=1 ./..."
+
+        #Run the example
+        #Here, we're not using a BackgroundThreadServer (i.e., CompatExample)
+        #because the unit tests are run synchronously
+        node = self.redpanda.nodes[0]
+        result = node.account.ssh_output(cmd).decode().split('\n')
+
+        failed = False
+        i = 0
+        while not failed and i < len(result):
+            failed = 'FAIL' in result[i].rstrip()
+            i += 1
+
+        assert not failed, "sarama unit tests failed"
+
+    # run sarama interceptors example
+    def test_sarama_interceptors(self):
+        def event_found(line):
+            return 'SpanContext' in line
+
+        #Setup and run the example
+        EXAMPLE_DIR = os.path.join(SaramaTest.TESTS_DIR, "examples/interceptors")
+        cmd = f"interceptors -brokers {self.redpanda.brokers()} -topic {self.topic}"
+        cmds = (os.path.join(EXAMPLE_DIR, cmd), "interceptors")
+
+        self._example.set_example(cmds, event_found)
+        self._example.start()
+
+        #Let the example run for a few seconds
+        time.sleep(10)
+
+        assert self._example.ok(), "sarama interceptors test failed"
+
+    # Run sarama http server example
+    def test_sarama_http_server(self):
+        def server_running(line):
+            return 'Listening for requests' in line
+
+        #Setup and run the example
+        EXAMPLE_DIR = os.path.join(SaramaTest.TESTS_DIR, "examples/http_server")
+        cmd = f"http_server -addr n1:8080 -brokers n1:9092"
+        cmds = (os.path.join(EXAMPLE_DIR, cmd), "http_server")
+        node = self.redpanda.nodes[0]
+
+        self._example.set_example(cmds, server_running, node)
+        self._example.start()
```

Comment:
I don't think this is quite the way to use the background service. The service should be getting its own node assigned by the ducktape orchestrator automatically.

I'd recommend removing the `set_example` interface. It looks like it exists because you are sharing the CompatExample instance across all the tests. But instead, you should be able to create a new instance in each test method, and that instances of the CompatExample will be assigned its own node.